src/log.rs

```diff
@@ -534,6 +574,14 @@ impl Log {
 				read_next = true;
 			}
 
+			// [Review] this clear reading file: but we
+			// do not know that mmap(s) got flushed before doing it.
+			// (I would keep it until flush, but do not know how to
+			// listen to mmap flush).
+			// Guess we could ensure buy usingf MAP_SYNC MAP_SHARED_VALIDATE.
+			// -> mmap2 is oppening MAP_SHARED so no MAP_SYNC at this point.
+			// But still MAP_SYNC looks a bit frightening, quote from https://lwn.net/Articles/731706/:
+			// `The result is a relatively simple mechanism that will perform far better than the currently available alternative — manually calling fsync() before each write to persistent memory. The potential downside is that any write operation can now create a flurry of I/O as the filesystem flushes out dirty metadata. That can cause the process to block in what was supposed to be a simple memory write, introducing latency that may be unexpected and unwanted.`
```

Comment:
The important bit is not to delete the `reading` file until flushing is complete. I suppose we could simply synchronously call `Mmap::flush` and `File::sync_data` for all indexes/tables here. It should not block other IO. 
`MAP_SYNC` does not look right indeed.. We don't want each write to be synced immediately, but rather sync at transaction, or even log file boundary.